theme/management/commands/report_resource_size_vs_quota_inconsistency.py

```diff
@@ -0,0 +1,162 @@
+                    'django': converted_total_size_django,
+                    'irods': used_value_irods_dz}
+                quota_report_list.append(report_dict)
+                print('quota incosistency: {} reported in django vs {} reported in iRODS for user {}'.format(
+                    converted_total_size_django, used_value_irods_dz, user.username), flush=True)
+
+                if update:
+                    print("Attempting to fix the inconsistency by updating file sizes in django")
+                    res_files = []
+                    for res in held_resources:
+                        print(f"Total files in resource {res.short_id}: {res.files.all().count()}")
+                        print(f'{current_site}/resource/{res.short_id}: currently {res.size} bytes')
+                        file_counter = 0
+                        # exclude files with size 0 as they don't exist in iRODS
+                        for res_file in res.files.exclude(_size=0).iterator():
+                            # this is an expensive operation (3 irods calls per file) - about 1 min for 100 files
+                            # size, checksum and modified time are obtained from irods and assigned to
+                            # relevant fields of the resource file object
+                            res_file.set_system_metadata(resource=res, save=False)
+                            res_files.append(res_file)
+                            file_counter += 1
+                            print(f"Updated file count: {file_counter}")
+                            if res_file._size <= 0:
+                                print(f"File {res_file.short_path} was not found in iRODS.")
+
+                        if res_files:
+                            ResourceFile.objects.bulk_update(res_files,
+                                                             ResourceFile.system_meta_fields(), batch_size=_BATCH_SIZE)
+                            print(f"Updated {file_counter} files for resource {res.short_id}")
+                            res.refresh_from_db()
+                            print(f'{current_site}/resource/{res.short_id}: now {res.size} bytes')
+                        else:
+                            print(f"Resource {res.short_id} contains no files.")
+
+                elif reset:
+                    print("Resetting file size cache in django")
+                    for res in held_resources:
+                        print(f'Resetting all filesizes in {current_site}/resource/{res.short_id}')
+                        res.files.update(_size=-1)
+                else:
+                    print('No action taken. Use --update or --reset to fix inconsistencies')
+
+        if quota_report_list:
+            with open(options['output_file_name_with_path'], 'w') as csvfile:
+                w = csv.writer(csvfile)
+                fields = [
+                    'User',
+                    'Usage of summed resource sizes in Django',
```

Comment:
This usage we are reporting here is before any update/reset of any of the resources. May be change the column heading accordingly?

Reply:
Thanks, yes.